variables.tf

```diff
@@ -87,146 +119,181 @@ variable "worker" {
 }
 
 variable "private_key" {
+    type        = string
     description = "content of private ssh key"
     # if empty string will read contents of file at var.private_key_file
-    default = ""
+    default     = ""
 }
 
 variable "public_key" {
+    type        = string
     description = "Public key"
     # if empty string will read contents of file at var.public_key_file
     default     = ""
 }
 
 variable "rhel_subscription_username" {
+    type    = string
     default = ""
 }
 
 variable "rhel_subscription_password" {
+    type    = string
     default = ""
 }
 
 variable "rhel_smt" {
+    type        = number
     description = "SMT value to set on the bastion node. Eg: on,off,2,4,8"
-    default = 4
+    default     = 4
 }
 
 ################################################################
 ### Instrumentation
 ################################################################
 variable "ssh_agent" {
+    type        = bool
     description = "Enable or disable SSH Agent. Can correct some connectivity issues. Default: false"
     default     = false
 }
 
 variable "installer_log_level" {
+    type        = string
     description = "Set the log level required for openshift-install commands"
-    default = "info"
+    default     = "info"
 }
 
 variable "helpernode_repo" {
+    type        = string
     description = "Set the repo URL for using ocp4-helpernode"
     # Repo for running ocp4 installations steps.
-    default = "https://github.com/RedHatOfficial/ocp4-helpernode"
+    default     = "https://github.com/RedHatOfficial/ocp4-helpernode"
 }
 
 variable "helpernode_tag" {
+    type        = string
     description = "Set the branch/tag name or commit# for using ocp4-helpernode repo"
     # Checkout level for https://github.com/RedHatOfficial/ocp4-helpernode which is used for setting up services required on bastion node
-    default = "dd8a0767c677fc862e45b6d70e5d04656ced5d28"
+    default     = "dd8a0767c677fc862e45b6d70e5d04656ced5d28"
 }
 
 variable "install_playbook_repo" {
+    type        = string
     description = "Set the repo URL for using ocp4-playbooks"
     # Repo for running ocp4 installations steps.
-    default = "https://github.com/ocp-power-automation/ocp4-playbooks"
+    default     = "https://github.com/ocp-power-automation/ocp4-playbooks"
 }
 
 variable "install_playbook_tag" {
+    type        = string
     description = "Set the branch/tag name or commit# for using ocp4-playbooks repo"
     # Checkout level for https://github.com/ocp-power-automation/ocp4-playbooks which is used for running ocp4 installations steps
-    default = "6adcde2b1cab38d52441a6cd54023869f569884e"
+    default     = "6adcde2b1cab38d52441a6cd54023869f569884e"
 }
 
 variable "ansible_extra_options" {
+    type        = string
     description = "Extra options string to append to ansible-playbook commands"
     default     = "-v"
 }
 
 variable "pull_secret_file" {
+    type      = string
     default   = "data/pull-secret.txt"
 }
```

Comment:
We should add a validation for having a non empty file as secret .

Reply:
This can be achieved using TF functions "fileexists" and "file".